maths/average_median.py

```diff
@@ -14,22 +16,19 @@ def median(nums):
         Median.
     """
     sorted_list = sorted(nums)
-    med = None
-    if len(sorted_list) % 2 == 0:
-        mid_index_1 = len(sorted_list) // 2
-        mid_index_2 = (len(sorted_list) // 2) - 1
-        med = (sorted_list[mid_index_1] + sorted_list[mid_index_2]) / float(2)
-    else:
-        mid_index = (len(sorted_list) - 1) // 2
-        med = sorted_list[mid_index]
-    return med
+    length = len(sorted_list)
+    mid_index = length >> 1
```

Comment:
Are we only using `length`once?  If yes,
```suggestion
    mid_index = len(sorted_list) >> 1
```